XamarinCommunityToolkit/ObjectModel/ObservableRangeCollection.shared.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Collections.Specialized;
+using System.ComponentModel;
+
+namespace Xamarin.CommunityToolkit.ObjectModel
+{
+	/// <summary>
+	/// Represents a dynamic data collection that provides notifications when items get added, removed, or when the whole list is refreshed.
+	/// </summary>
+	/// <typeparam name="T"></typeparam>
+	public class ObservableRangeCollection<T> : ObservableCollection<T>
+	{
+
+		/// <summary>
+		/// Initializes a new instance of the System.Collections.ObjectModel.ObservableCollection(Of T) class.
+		/// </summary>
+		public ObservableRangeCollection()
+			: base()
+		{
+		}
+
+		/// <summary>
+		/// Initializes a new instance of the System.Collections.ObjectModel.ObservableCollection(Of T) class that contains elements copied from the specified collection.
+		/// </summary>
+		/// <param name="collection">collection: The collection from which the elements are copied.</param>
+		/// <exception cref="System.ArgumentNullException">The collection parameter cannot be null.</exception>
+		public ObservableRangeCollection(IEnumerable<T> collection)
+			: base(collection)
+		{
+		}
+
+		/// <summary>
+		/// Adds the elements of the specified collection to the end of the ObservableCollection(Of T).
+		/// </summary>
+		public void AddRange(IEnumerable<T> collection, NotifyCollectionChangedAction notificationMode = NotifyCollectionChangedAction.Add)
```

Comment:
From my point of view, it would be slightly better to prevent ArgumentException by defining own enum with two options "Add" or "Reset" rather than allowing to pass "Remove" with getting an exception